core/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java

```diff
@@ -36,6 +38,20 @@ public ActionListenerResponseHandler(ActionListener<Response> listener) {
         this.listener = listener;
     }
 
+    /**
+     * Convenience method for delegating a response provided by supplier to the provided transport channel
+     */
+    public static <T extends TransportResponse> ActionListenerResponseHandler responseHandler(ActionListener<T> listener, Supplier<T>
```

Comment:
How about instead adding response supplier to the constructor and making the class non-abstract (now that we are in Java8 land)? Currently every consumer of the class has to implement the responeSupplier anyhow.

```
private final ActionListener<Response> listener;
private final Supplier<Response> responseSupplier;

public ActionListenerResponseHandler(ActionListener<Response> listener, Supplier<Response> responseSupplier) {
    this.listener = listener;
    this.responseSupplier = responseSupplier;
}

@Override
public Response newInstance() {
    return responseSupplier.get();
}
```

Usage sites look simpler:

```
new ActionListenerResponseHandler(listener, action::newResponse)
```

instead of 

```
new ActionListenerResponseHandler(listener) {
            @Override
            public TransportResponse newInstance() {
                return action.newResponse();
            }

        })
```